manifests/plugin/container_gateway.pp

```diff
@@ -22,8 +34,14 @@
   Boolean $enabled = true,
   Foreman_proxy::ListenOn $listen_on = 'https',
   Stdlib::HTTPUrl $pulp_endpoint = "https://${facts['networking']['fqdn']}",
+  String $database_backend = 'postgres',
   Stdlib::Absolutepath $sqlite_db_path = '/var/lib/foreman-proxy/smart_proxy_container_gateway.db',
   Optional[Integer] $sqlite_timeout = undef,
+  Stdlib::Host $postgresql_host = 'localhost',
+  Stdlib::Port $postgresql_port = 5432,
+  String $postgresql_database = 'container_gateway',
+  String $postgresql_user = 'foreman-proxy',
```

Comment:
I'm surprised we don't have a variable for the username yet. It would be good to have that.

Reply:
I think I finally realized what you were getting at here -- we need to match the postgres user up with whatever the user for the smart proxy is (since it might not be foreman-proxy).

Reply:
For pulpcore it looks like we define the name as such: https://github.com/theforeman/puppet-pulpcore/blob/21aa39e28f19b40bf328ebba514465fb9cfb9fe2/manifests/init.pp#L224

Reply:
Based on https://github.com/theforeman/puppet-foreman_proxy/blob/5da646fd836040a685e4669e14eb76519b3e5224/manifests/params.pp#L14, it looks like I could use `$foreman_proxy::globals::user`